readthedocs/api/v3/views.py

```diff
@@ -126,6 +130,20 @@ class ProjectsViewSet(APIv3Settings, NestedViewSetMixin, ProjectQuerySetMixin,
         'active_versions.last_build.config',
     ]
 
+    def get_serializer_class(self):
```

Comment:
I see what you're doing here, but stylistically, I think it would be better to get rid of this method and make `create()` explicitly use a different serializer on incoming POST data.

In all cases, you want the JSON output to use `ProjectSerializer`. However, when there's user input like a user creating a project they aren't going to send every last field as some of them are automatic or have reasonable defaults so a different serializer may make sense. Since you're already overriding the output serializer in `create()`, I'd rather you just instead overrode the input serializer and used the default output serializer.

Reply:
Actually, it looks like this is necessary due to how `CreateModelMixin` works.

Reply:
I changed this. Although, since this PR is based on the "Import Project" one, some things changed there and I will re-check this once that one gets merged.